viz/charts/linkerd-viz/templates/tap-injector-rbac.yaml

```diff
@@ -0,0 +1,95 @@
+###
+### Tap Injector RBAC
+###
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: {{.Values.namespace}}-tap-injector
+  labels:
+    {{.Values.extensionAnnotation}}: {{.Values.namespace}}
+    component: tap-injector
+rules:
+- apiGroups: [""]
+  resources: ["namespaces"]
+  verbs: ["get", "list"]
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: {{.Values.namespace}}-tap-injector
+  labels:
+    {{.Values.extensionAnnotation}}: {{.Values.namespace}}
+subjects:
+- kind: ServiceAccount
+  name: tap-injector
+  namespace: {{.Values.namespace}}
+roleRef:
+  kind: ClusterRole
+  name: {{.Values.namespace}}-tap-injector
+  apiGroup: rbac.authorization.k8s.io
+---
+kind: ServiceAccount
+apiVersion: v1
+metadata:
+  name: tap-injector
+  namespace: {{.Values.namespace}}
+  labels:
+    {{.Values.extensionAnnotation}}: {{.Values.namespace}}
+    component: tap-injector
+    namespace: {{.Values.namespace}}
+{{- include "partials.image-pull-secrets" .Values.imagePullSecrets }}
+---
+{{- $host := printf "tap-injector.%s.svc" .Values.namespace }}
+{{- $ca := genSelfSignedCert $host (list) (list $host) 365 }}
+{{- if (not .Values.tapInjector.externalSecret) }}
+kind: Secret
+apiVersion: v1
+metadata:
+  name: tap-injector-k8s-tls
+  namespace: {{ .Values.namespace }}
+  labels:
+    {{.Values.extensionAnnotation}}: {{.Values.namespace}}
+    component: tap-injector
+    namespace: {{.Values.namespace}}
+  annotations:
+    {{.Values.createdByAnnotation}}: {{default (printf "linkerd/helm %s" .Values.linkerdVersion) .Values.cliVersion}}
```

Comment:
I think this needs to be added to all the other resources in this file.

Reply:
I think I may just keep it on this secret. Pretty much all the RBAC templates in the Viz extension and control plane have this annotation for their secrets, but not other resources like cluster role and bindings.

If we want this annotation everywhere I'd rather make a change where we add it everywhere, instead of semi-randomly doing it.